source/testing/index.md

```diff
@@ -1,79 +1,146 @@
+* A helper computes the current date properly.
+
+We can create these type of isolated tests which neither require the application's container to be setup
+nor any user interaction with the `setupTest` helper as well.
+
 ### Testing Frameworks
 
 [QUnit] is the default testing framework for this guide, but others are supported through third-party addons.
 
+### Testing Blueprints
+
+By default whenever you are creating a new component, helper, service, or any another module in your app,
+[Ember CLI](https://ember-cli.com/generators-and-blueprints) will automatically create a QUnit-based test file
+- a testing blueprint - for you.
+This blueprint will contain all the basic test setup necessary for testing the module you have just created
+and help you get started writing your first test straight away.
+
+Imagine we'd create a new `location` service by running `ember generate service location`.
+If we have a look at the newly created blueprint under `tests/unit/services/location-test.js` we'd find the following:
+
+```tests/unit/models/some-thing-test.js
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+
+module('Unit | Service | location', function(hooks) {
+  setupTest(hooks);
+
+  // Replace this with your real tests.
+  test('it exists', function(assert) {
+    let service = this.owner.lookup('service:location');
+    assert.ok(service);
+  });
+});
+```
+
+Note how this test setup follows a nested structure.
+Each test case will be wrapped by a [QUnit.module](https://api.qunitjs.com/QUnit/module) function
+which allows us to scope several tests into a group.
+We can now apply the same testing setup to all tests in this test group easily,
+or we can run those tests independently from the rest of the test suite.
+
 ### How to Run Your Tests
 
 Run your tests with `ember test` on the command-line. You can re-run your tests on every file-change with `ember test --server`.
 
-Tests can also be executed when you are running a local development server (started by running `ember server`), at the `/tests` URI which renders the `tests/index.html` template.
+Tests can also be executed when you are running a local development server (started by running `ember server`),
+at the `/tests` URI which renders the `tests/index.html` template.
 A word of caution using this approach:
-Tests run using `ember server` have the environment configuration `development`, whereas tests executed under `ember test --server` are run with the configuration `test`.  This could cause differences in execution, such as which libraries are loaded and available.  Therefore its recommended that you use `ember test --server` for test execution.
+Tests run using `ember server` have the environment configuration `development`,
+whereas tests executed under `ember test --server` are run with the configuration `test`.
+This could cause differences in execution, such as which libraries are loaded and available.
```

Comment:
Should we even mention the `ember server` and `/tests` approach? I think we shouldn't, as it's considered poor practice (like we mention here). Let's go ahead and ✂️ that discussion completely

Reply:
Where is the recommendation that `ember server` is a poor practice? I read it as a good practice in this section actually:

> Therefore its recommended that you use `ember test --server` for test execution.